src/main.js

```diff
@@ -1,22 +1,264 @@
 /// <reference types="jquery" />
 
-//1. Construir un pokedex
-// a.buscar por id
-// b.buscar por tipo
-
-
-
-
-
-
-
+document.querySelector('#search-pokemon-button').addEventListener('click', function () {
+    const pokemon = $('#pokemon-id-input').val().toLowerCase();
+    getPokemonHandler(pokemon)
 
+})
+document.querySelector('#previous-page').addEventListener('click', function () {
+    const PREVIOUS_PAGE_OFFSET = 16;
+    const previousPage = firstPokemonOnListId - PREVIOUS_PAGE_OFFSET;
+    getPokemonList(previousPage)
+})
+document.querySelector('#next-page').addEventListener('click', function () {
+    const nextPage = lastPokemonOnListId;
+    getPokemonList(nextPage)
+})
+$('#pokemon-list').on('click', event => {
+    const pokemon = event.target.id;
+    if(!event.target.classList.contains("btn-link")) return;
+    getPokemonHandler(pokemon)
+})
+getPokemonList();
 
+/**
+ * Fetches a list of Pokemon with the given offset and limit.
+ * @param {number} [offset=0] - The offset of the first Pokemon to fetch.
+ * @param {number} [amount=10] - The maximum number of Pokemon to fetch.
+ */
+function getPokemonList(offset = 0, amount = 15){
+    const LOWEST_POKEMON_OFFSET = 0;
+    offset = Math.max(offset, LOWEST_POKEMON_OFFSET);
+    fetch(`https://pokeapi.co/api/v2/pokemon/?offset=${offset}&limit=${amount}`)
+    .then(response => response.json())
+    .then(data => {
+        if(!totalPokemon) { updateTotalPokemon(data.count); };
+        updateFirstAndLastPokemon(offset, amount);
+        updatePokemonList(data.results, offset);
+    })
+    .catch(() => {
+        alert('Could not get pokemons list');
+    })
+}
```

Comment:
I think this function could only get the Pokemon list and not update it, handling the updates in another function? Also, you could name the function as `getPokemons` instead of `getPokemonList`, the plural implicitly says that is a list of pokemons.

Reply:
The noun Pokemon is uncountable. The plural form of Pokemon is also Pokemon.